clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -303,6 +318,20 @@ public PartitionConsumptionState(
     this.leaderCompleteState = LeaderCompleteState.LEADER_NOT_COMPLETED;
     this.lastLeaderCompleteStateUpdateInMs = 0;
     this.pendingReportIncPushVersionList = offsetRecord.getPendingReportIncPushVersionList();
+
+    KeyUrnCompressionDict keyUrnCompressionDict = offsetRecord.getKeyUrnCompressionDict();
+    if (keyUrnCompressionDict != null) {
+      if (keyUrnCompressionDict.keyUrnCompressionDictionaryVersion != 1) {
+        throw new VeniceException(
+            "Unsupported key urn compression dictionary version: "
+                + keyUrnCompressionDict.keyUrnCompressionDictionaryVersion);
+      }
+      UrnDictV1 urnDictV1 = UrnDictV1.loadDict(keyUrnCompressionDict.keyUrnCompressionDictionary);
+      this.keyUrnCompressor = new KeyUrnCompressor(
+          keySchema,
+          keyUrnCompressionDict.keyUrnFields.stream().map(Object::toString).collect(Collectors.toList()),
```

Comment:
How does it work with the key schema that's primitive string?

Reply:
If the key schema is string, you don't need to pass anything urn fields when using this feature.